the-collab-lab/tcl-22-smart-shopping-list
# Associate form data with userToken & capture frequency of purchase

## Description

Adding functionality to `AddGroceries` component so that form is able to capture the frequency of purchase for each item the user adds.

Form will include:
* Name of item
* How soon are you likely to buy it again?
  * Soon (in the next 7 days)
  * Kind of soon (in the next 14 days)
  * Not soon (in the next 30 days)
* Last purchased date

## Related Issue


Closes #5 

## Acceptance Criteria


* User is presented with a form that allows them to enter the name of the item and select how soon they anticipate needing to buy it again (Soon, Kind of Soon, Not Soon)
* When the user submits the form, the item is saved to the database, associated with the user’s token
* Along with the item name, an integer corresponding to the estimated number of days until next purchase is saved: `7` for “Soon”, `14` for “Kind of Soon”, and `30` for “Not Soon”
* The last purchased date should be set to `null` initially (i.e. you can create an item without purchasing it)
* Item names should be displayed as a simple, unordered list on the list view after they’re added

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

## Updates

### Before


![image](https://user-images.githubusercontent.com/58532369/114472958-48f5e500-9bc1-11eb-9697-d53d890565ae.png)

### After


![Screen Cast 2021-04-14 at 7 28 53 PM](https://user-images.githubusercontent.com/58532369/114793150-c05d7d00-9d57-11eb-8772-b32cb84dee55.gif)


## Testing Steps / QA Criteria


To test the feature, use one of the following ways:
1) Follow the link to see the preview: https://tcl-22-shopping-list--pr20-kk-cf-new-item-2lpzshyj.web.app
2) Launch the app locally from your terminal:
 • Pull down this branch with `git pull origin kk-cf-new-item` and check that branch out with `git checkout origin kk-cf-new-item`
 • Run `npm start`  to launch the app in your browser at [http://localhost:3000](http://localhost:3000)
When on application, click on 'Add Groceries' link in navbar to see user form. Fill user form and click 'Submit,' followed by navigating to 'Grocery List' to see updated list.
